src/index.ts

```diff
@@ -60,7 +60,7 @@ export default class Machine extends erx.Bus<StateName> {
     this.transitioning = false;
   }
 
-  goTo(next: StateName, data?: any): boolean {
+  goTo(next: StateName, getState?: Function): boolean {
```

Comment:
Please add the function signature `() => state.MachineState<A>`

Reply:
Hmmmm, the machineState lives on cyan under core/state/index.js. Shall i import from there? or move the types over to here? Or the whole thing!?

Reply:
Hmm, that is a really good point...

Reply:
Okay, how about `goTo(next: StateName, getState?: () => S)` and add the FSM<S> type in the class generic declaration.

And then, back in `src/core/index.ts`, you define FSM as FSM<state.MachineState>, and... Yea, same in `src/index.ts`